### PR TITLE
Implement Pause/Resume Button Functionality

### LINKED ISSUE #18: Add in Pause/Resume button + functionality
pretty much taken care of with #15, logging an issue for traceability

### PR DESCRIPTION
**Pull request by ColemanMitch**

### CODE REVIEW

File: backend/server.js [hunk 1]
```diff
@@ -1,48 +1,55 @@
-let express = require('express')
-let request = require('request')
-let querystring = require('querystring')
+let express = require("express");
```
Comment (dwilliams27) [inline backend/server.js:1]: New linter? Or did you manually change all dat

File: src/Timer.tsx [hunk 2]
```diff
@@ -45,24 +36,35 @@ class Timer extends React.Component<TimerProps, TimerState> {
...
-      newTime.minutes -= 1;
-      newTime.seconds = 60;
+      newTime.seconds -= 1;
+      this.setState({
+        time: newTime
+      });
+    }
+    else { // if it's paused make sure ticking is false
+      this.setState({
+        ticking: false
```
Comment (dwilliams27) [inline src/Timer.tsx]: Is ticking ever used logically differently than paused? Or could ticking be removed and we only refer to props.paused instead?

File: backend/server.js (hunk 1 above)
Comment (ColemanMitch) [reply backend/server.js:1]: New linter that auto formats .js files

File: src/Timer.tsx (hunk 2 above)
Comment (ColemanMitch) [reply src/Timer.tsx]: We would want to  stop ticking all together once we actually reach the max song count. 

That hasn't been implemented yet, so lemme log that as another potential issue.

File: src/App.tsx [hunk 3]
```diff
@@ -128,6 +131,33 @@ class App extends Component<void, AppState> {
     });
   }
 
+  pauseCurrentPlayback(): void {
```
Comment (dwilliams27) [inline src/App.tsx:172]: Could probably combine these 2 methods into 1 like:

```
adjustPlayback(paused: boolean): void {
   this.setState({paused})
    (paused ? this.spotifyService.pauseCurrentPlayback() : this.spotifyService.resumeCurrentPlayback()).then(res => {
      res.body?.getReader().read().then(body => {
        if(body?.done) {
          setTimeout(() => this.fetchCurrentlyPlaying(), 1000);
        }
      });
    });
  }
```

File: src/Timer.tsx [hunk 4]
```diff
@@ -45,24 +36,35 @@ class Timer extends React.Component<TimerProps, TimerState> {
   }
 
   tick = async () => {
+    console.log(this.props, this.state);
```
Comment (dwilliams27) [inline src/Timer.tsx]: Assuming WIP, but probably dont want to print the state every tick lol

File: src/Timer.tsx [hunk 5]
```diff
@@ -45,24 +36,35 @@ class Timer extends React.Component<TimerProps, TimerState> {
...
-      if(this.state.time.minutes === 0) {
-        this.props.skipToNextSong();
-        this.setState({
-          time: this.createTimeObj(this.props.interval),//(START_TIME),
-          songCount: this.state.songCount + 1,
-          ticking: false
-        });
-        return;
+    if (!this.props.paused) { // if it's playing (not paused)
+        if(this.state.time.seconds === 0) {
```
Comment (dwilliams27) [inline src/Timer.tsx]: Minor, but indentation a bit off here

File: src/App.tsx (hunk 3 above)
Comment (dwilliams27) [reply src/App.tsx:172]: Could even make it not have any parameters and just invert what's currently the state of `paused` 🤔

File: src/Timer.tsx (hunk 4 above)
Comment (ColemanMitch) [reply src/Timer.tsx]: Good catch

File: src/App.tsx (hunk 3 above)
Comment (ColemanMitch) [reply src/App.tsx:172]: You thinking something like?

```this.setState({ paused: !paused})```

**Comment by ColemanMitch** [comment]
Major of diffs are from `package-lock.json`. Ready for re-review whenever @dwilliams27

File: src/App.tsx (hunk 3 above)
Comment (dwilliams27) [reply src/App.tsx:172]: Yeah could go with 
```
invertPlayback(): void {
   
    (!this.state.paused ? this.spotifyService.pauseCurrentPlayback() : this.spotifyService.resumeCurrentPlayback()).then(res => {
      res.body?.getReader().read().then(body => {
        if(body?.done) {
          setTimeout(() => this.fetchCurrentlyPlaying(), 1000);
        }
      });
    });
    this.setState({pause: !this.state.paused})
  }
  ```
  
 Though doesn't really matter, up to you my man

**Review by dwilliams27 — APPROVED** [review]
LGTM, up to you about changing the pause/play methods. You're good to merge it whenever 👍

**Comment by dwilliams27** [comment]
Actually you may want to remove the changes to the backend from this PR, I cleaned it up and reworked it a bit in [this PR](https://github.com/ColemanMitch/Spowerfy-Redux/pull/25).

**Comment by ColemanMitch** [comment]
> Actually you may want to remove the changes to the backend from this PR, I cleaned it up and reworked it a bit in [this PR](https://github.com/ColemanMitch/Spowerfy-Redux/pull/25).

~~Shweet, I can revert the changes to `server.js` since they were just linting 😎~~

Actually wait nvm. We can just keep as is since they're just linting changes

**Pull request merged**